### PR TITLE
Restructure backend for pagination

### PR DESCRIPTION
**Pull request by Huy-Ngo**
In this PR, I restructure backend for pagination-oriented to fit more with the frontend design, so each request is `source/<page>` instead of `source/<n_images>` like before.

I also:

- Filter out pdf files
- Add more page search for wiki
- Replace `_` with ` ` for titles of images crawled from wiki

### CODE REVIEW

File: backend/app.py [hunk 1]
```diff
@@ -9,79 +9,49 @@
...
-    """Get `n_images` images from Kaggle API."""
-    return {'data': randomize(kaggle_data, n_images)}
-
-
-@app.route('/isic/<int:n_images>')
-def get_isic(n_images):
-    """Get `n_images` images from ISIC API."""
-    return {'data': randomize(isic_data, n_images)}
-
+def get_page(data, page):
```
Comment (dinhanhx) [inline backend/app.py:18]: Do we have `page = 0`? Or our pages start from 1?

Comment (Huy-Ngo) [reply backend/app.py:18]: We're programmers. We start from 0.

Comment (dinhanhx) [reply backend/app.py:18]: Oh. I was confused with simple math.

**Review by dinhanhx — APPROVED** [review]
It works fine.

**Comment by Huy-Ngo** [comment]
Now it can crawl 380 images from wikimedia without the PDF.

**Pull request merged**